src/components/context/data/DataState.ts

```diff
@@ -0,0 +1,48 @@
+import { Instrument, MeasurementVariable } from 'cheminfo-types';
+
+export interface DataState {
+  measurements: Measurements;
+}
+
+interface Measurements {
+  ir: {
+    entries: MeasurementBase[];
+  };
+  raman: {
+    entries: MeasurementBase[];
+  };
+  uv: {
+    entries: MeasurementBase[];
+  };
+  nmr1h: {
+    entries: MeasurementBase[];
+  };
+  mass: {
+    entries: MeasurementBase[];
+  };
+  other: {
+    entries: MeasurementBase[];
+  };
+}
+
+interface MeasurementBase {
+  id: string;
+  title?: string;
+  instrument?: Instrument;
+  neta: Record<string, string>;
+  info: Record<string, string | number>;
+  data: {
+    variables: Record<string, MeasurementVariable>;
+  }[];
+}
+
+export type MeasurementKind = keyof Measurements;
+
+export const kindsLabel: Record<MeasurementKind, string> = {
```

Comment:
@wadjih-bencheikh18 How should I do so that all the properties from 'MeasurementKind' are defined in this object and that there is an error if there is one extra property or missing property.

Reply:
Using Record<MeasurementKind, string> is enough here

Reply:
I don't think that this solves the problem of a missing property.

Reply:
@targos could you comment on this ?

Reply:
I'm not sure what you mean. If I forget a property there is an error:
<img width="715" alt="image" src="https://user-images.githubusercontent.com/2352663/175480722-1b01bebc-06ca-4848-8bdf-382c3db994a1.png">